Library/Homebrew/linkage_checker.rb

```diff
@@ -0,0 +1,121 @@
+require "set"
+require "keg"
+require "formula"
+
+class LinkageChecker
+  attr_reader :keg, :formula
+  attr_reader :brewed_dylibs, :system_dylibs, :broken_dylibs, :variable_dylibs
+  attr_reader :undeclared_deps, :reverse_links
+
+  def initialize(keg, formula = nil)
+    @keg = keg
+    @formula = formula || resolve_formula(keg)
+    @brewed_dylibs = Hash.new { |h, k| h[k] = Set.new }
+    @system_dylibs = Set.new
+    @broken_dylibs = Set.new
+    @variable_dylibs = Set.new
```

Comment:
One prefixed with `@rpath/`, `@loader_path/`, or `@executable_path/`. Those are resolved by the dynamic linker and aren't immediately accessible to static analysis (what is done here in this class). FWIW, `dyld(1)` uses the same terminology.